### DISCUSSION TITLE
useWatch for nested register

### DISCUSSION THREAD

**Discussion by neighborhood999** [react-hook-form/react-hook-form#2661] [Q&A] [answered]
According to [document](https://react-hook-form.com/api#useForm) says, we can create nested form fields, e.g.:

```js
name="social.twitter[0]"
name="social.twitter[1]"
```

`social.twitter[0]` represent `account`, `social.twitter[1]` represent `status` (on or off).

In `useWatch` I can't watch `social.twitter` any value (I had added `ref={register}` to my compnent):

```js
const status = useWatch({
  control,
  name: 'social[twitter]',
});

console.log(status) // undefined
```

**Update**
Without providing `name` it will get all fields and I found can't get `social` related fields :(

**Comment by bluebill1049** [comment, marked as answer]
hey @neighborhood999,

Does it work with the key?
```
const status = useWatch({
  control,
  name: 'social',
});
```

**Reply by neighborhood999** [reply]
Sorry my fault, it works fine..., Thanks @bluebill1049!
